mandatory/2-function-creation.js

```diff
@@ -1,160 +1,108 @@
 /*
-  Complete the function to return a boolean describing whether a user is acceptable. 
-  To be acceptable, two conditions must be fulfilled
-    1. the user should be 18 or older
-    2. the user must be logged in
+Complete the function to check if the variable `num` satisfies the following requirements:
+- is a number
+- is even
+- is less than or equal to 100
+Tip: use logical operators
 */
-function isAcceptableUser(userAge, isLoggedIn) {}
 
-/*
-  Complete the function to apply discount percent based on how much is totalPrice in user cart.
-    - When the total price is greater than 200, a 10% discount should be applied
-    - When the total price is less than 200, a 5% discount should be applied
-
-  The function should return the new price to be paid (e.g. if the totalPrice is 150, a 5% discount
-    is applieds and 142.5 should be returned)
-*/
-
-function applyDiscount(totalPrice) {}
+function validate(num) {
+  //The typeof operator returns a string indicating the type of the operand's value.
+  if (typeof num == "number" && num % 2 == 0 && num <= 100) {
+    return true;
+  } else {
+    return false;
+  }
+}
 
 /*
-  Complete the function to print to the console the odd numbers between 1 and limit (use a while loop):
-  */
-function printOddNumbers(limit) {}
-
-/*
-  Complete the buyTwoGetTheCheapestFree function: if user buys two items, the cheapest item will be free!
-  The function should return the price to be paid once the discount is applied.
+Write a function that:
+- takes a number as input
+- return a string formatted as percentages (e.g. 10 => "10%")
+- the number must be rounded to 2 decimal places
+- numbers greater 100 must be replaced with 100
 */
-function buyTwoGetTheCheapestFree(price1, price2) {}
 
-/*
-  Complete the function to determine if it is suitable for a person to register based on their age!
-  - if the person is 12 or younger it should return "You Are Too Young To Register"
-  - if the person is older than 12 and younger than 90 it should return "You Can Register"
-  - if the person is 90 or older it should return "You Don't Need To Register"
-*/
-function canRegister(age) {}
+function formatPercentage(num) {
+  //num = Math.min(num, 100); // replace number with 100 if it's greater than 100
+  if (num >= 100) {
+    num = 100;
+  }
+  return num.toFixed(2).replace(/\.?0+$/, "") + "%";
```

Comment:
This seems overly complicated. 
You can rewrite this with `Math.round` which will not produce the trailing zeros you are removing. 

```
return `${Math.round(num * 100) / 100}%`
```
will produce the correct answer and you do not need to use regex. But if you want to use regex, it's always best to leave a comment explaining what it does becaue regex is not easy to understand